backend/adapter_processor_v2/serializers.py

```
@@ -24,10 +26,19 @@ class TestAdapterSerializer(serializers.Serializer):
+    # ``shared_groups`` is no longer an M2M on AdapterInstance — declare it
+    # explicitly so ``fields = "__all__"`` continues to expose it. Share
+    # mutations go through ``POST /adapter/{id}/share/`` (UN-2977 plan §B).
+    shared_groups = serializers.PrimaryKeyRelatedField(many=True, read_only=True)
+
     class Meta:
         model = AdapterInstance
         fields = "__all__"
+        extra_kwargs = {
+            "shared_users": {"read_only": True},
+            "shared_to_org": {"read_only": True},
+        }
```

Ruff (0.15.13)

[warning] Mutable default value for class attribute

(RUF012)